spark-jobs/src/main/scala/za/co/absa/enceladus/conformance/interpreter/DynamicInterpreter.scala

```diff
@@ -309,7 +308,10 @@ case class DynamicInterpreter(implicit inputFs: FileSystem) {
     val mappingTableDef = ictx.dao.getMappingTable(rule.mappingTable, rule.mappingTableVersion)
     val mappingTablePath = PartitioningUtils.getPartitionedPathName(mappingTableDef.hdfsPath,
       ictx.progArgs.reportDate)
-    val mappingTableSize = HadoopFsUtils.getOrCreate(inputFs).getDirectorySizeNoHidden(mappingTablePath)
+    //accommodate different fs for the mapping table or different bucket
+    val mappingTableFs = PathWithFs.fromPath(mappingTablePath)(ictx.spark.sparkContext.hadoopConfiguration)
+
+    val mappingTableSize = HadoopFsUtils.getOrCreate(mappingTableFs.fileSystem).getDirectorySizeNoHidden(mappingTableFs.path)
```

Comment:
I would consider making this a function in the `HadoopFSUtils` object, only taking in the path as a `String` and `implcit HadoopConfiguration`.

Reply:
I think something like this would indeed make sense on a larger refactoring, but this way we would need to pass the path 2 times: once when instantiating the object and then when calling `getDirectorySizeNoHidden`. Is it good enough for now?

Reply:
Not sure, I understand the double passing of path. But sure enough good enough now (as the approval suggests)